src/main/java/com/esri/arcgisruntime/toolkit/FloorFilter.java

```diff
@@ -0,0 +1,715 @@
+                Logger logger = Logger.getLogger(FloorFilter.class.getName());
+                logger.warning("The FloorManager failed to load with error: " + floorManager.getLoadError().getCause());
+              }
+            });
+            // load the floor manager if it is not already loaded
+            if (floorManager.getLoadStatus() != LoadStatus.LOADED) {
+              floorManager.loadAsync();
+            }
+          } else {
+            // if not floor aware set the floor manager property to null
+            floorManagerProperty.set(null);
+            Logger logger = Logger.getLogger(FloorFilter.class.getName());
+            logger.info("The GeoModel attached to the provided GeoView is not Floor Aware. FloorFilter will not " +
+              "be displayed. Call FloorFilter.refresh() after updating the GeoModel to try again.");
+          }
+        } else if (geoModel.getLoadStatus() == LoadStatus.FAILED_TO_LOAD) {
+          Logger logger = Logger.getLogger(FloorFilter.class.getName());
+          logger.warning("The GeoModel failed to load with error: " + geoModel.getLoadError().getCause());
+        }
+      });
+      if (geoModel.getLoadStatus() != LoadStatus.LOADED) {
+        // load the geomodel if it is not already loaded
+        geoModel.loadAsync();
+      }
+    }
+  }
+
+  /**
+   * A method that can be called to refresh the data on the existing FloorFilter. For example, if the GeoModel attached
+   * to the GeoView is updated, the FloorManager can be reloaded. Selected properties will be reset to null.
+   *
+   * @since 100.14.0
+   */
+  public void refresh() {
+    setSelectedSite(null);
+    setSelectedFacility(null);
+    setSelectedLevel(null);
+    setupFloorManager();
+  }
+
+  /**
+   * If there is a selected floor level, the method sets all floor levels with the same vertical order to be visible,
+   * and sets the visibility of all other levels to false. If there is no selected level, no action is taken.
+   *
+   * @since 100.14.0
+   */
+  public void filterLevelsVisibility() {
+    if (getSelectedLevel() != null) {
+      getLevels().forEach(level -> level.setVisible(level.getVerticalOrder() == getSelectedLevel().getVerticalOrder()));
+    }
+  }
+
+  /**
+   * For all facilities, set the visibility of the floor level at vertical order zero, the default, to visible, and all
+   * other levels to false.
+   *
+   * @since 100.14.0
+   */
+  public void setLevelVisibilityToDefault() {
+    if (getFacilities() != null) {
+      getFacilities().forEach(facility -> facility.getLevels().forEach(level -> level.setVisible(level.getVerticalOrder() == 0)));
+    }
+  }
+
+  /**
+   * Set the visibility of all floor levels on the FloorManager to visible.
+   * This can be useful when viewing a floor aware ArcGISScene to see all floor levels in 3D.
+   *
+   * @since 100.14.0
+   */
+  public void setAllLevelsVisible() {
+    getLevels().forEach(level -> level.setVisible(true));
```

Comment:
Could this be a boolean property - true makes all levels visible? I'm not sure what false would do but there's probably some reasonable behaviour. As a property it would mean user code could bind a toggle or whatever to turn this on/off. Also a property would presumably make implementing a toggle on the skin easier.

Reply:
Thanks for raising this. I took it to the toolkit team as it was not something we'd discussed in depth - the idea of visibility being set externally. It's a really good point with it being so woven into the selection. We decided to add a line to the doc making it clear that the floor filter will change visibility:
`Please note: The FloorFilter will not respect any custom visibility settings applied to the FloorManager's contents outside of the FloorFilter, and can change a FloorLevel's visibility at any point.`
As such, like the other helper methods, I've no longer made this public to avoid confusion and any issues with binding. If a use case arises we would revisit the design.